core/fx/retry.go

```diff
@@ -1,31 +1,76 @@
 package fx
 
-import "github.com/zeromicro/go-zero/core/errorx"
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/zeromicro/go-zero/core/errorx"
+)
 
 const defaultRetryTimes = 3
 
+var timeoutErr = errors.New("retry timeout")
+
 type (
 	// RetryOption defines the method to customize DoWithRetry.
 	RetryOption func(*retryOptions)
 
 	retryOptions struct {
-		times int
+		times    int
+		interval time.Duration
+		timeout  time.Duration
 	}
 )
 
 // DoWithRetry runs fn, and retries if failed. Default to retry 3 times.
-func DoWithRetry(fn func() error, opts ...RetryOption) error {
+// Note that if the fn function accesses global variables outside the function and performs modification operations,
+// it is best to lock them, otherwise there may be data race issues
+func DoWithRetry(fn func(ctx context.Context) error, opts ...RetryOption) error {
```

Comment:
Don't change the original API, add a new method like `DoWithRetryCtx`.

Otherwise, it's a breaking change.

Reply:
A DowithRetryCtx method has been added, and the original method will not be affected